crates/turborepo-lockfiles/src/pnpm/data.rs

```diff
@@ -252,6 +251,27 @@ impl PnpmLockfile {
         }
         Ok(pruned_patches)
     }
+
+    // Create a projection of all fields in the lockfile that could affect all
+    // workspaces
+    fn global_fields(&self) -> GlobalFields {
+        GlobalFields {
+            version: &self.lockfile_version.version,
+            checksum: self.package_extensions_checksum.as_deref(),
+            overrides: self.overrides.as_ref(),
+            patched_dependencies: self.patched_dependencies.as_ref(),
+            settings: self.settings.as_ref(),
+        }
+    }
```

Comment:
This is something I took from the global cache key as I find `self.global_fields() != other.global_fields()` far more legible than `self.lockfile_version.version != other.lockfile_version.version || self.package_extensions_checksum.as_deref() != ...`. Only opted to do it for `pnpm` since other lockfiles have 2 or fewer global fields.